project/Publish.scala

```diff
@@ -0,0 +1,32 @@
+package akka.grpc
+
+import sbt._, Keys._
+
+/**
+ * For projects that are not to be published.
+ */
+object NoPublish extends AutoPlugin {
+  override def requires = plugins.JvmPlugin
+
+  override def projectSettings = Seq(
+    publishArtifact := false,
+    publish := {},
+    publishLocal := {},
+    skip in publish := true,
+  )
+}
```

Comment:
All of this accumulated throughout the years. `skip in publish := true` should be the only one required to do the job. Lets try to adjust the folklore to only that setting. :)